### PR TITLE
Expandable content -block

### PR DESCRIPTION
**Pull request by Maijjay**

### CODE REVIEW

**Review by george-misan — APPROVED** [review]

File: services/course-material/src/components/ContentRenderer/moocfi/FlipCard/FlipCardBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            flex-direction: column;
+            justify-content: center;
+          `}
+        >
+          <ContentRenderer
+            data={[frontCard]}
+            editing={false}
+            selectedBlockId={null}
+            setEdits={function (): void {
+              throw new Error("Function not implemented.")
```
Comment (nygrenh) [inline services/course-material/src/components/ContentRenderer/moocfi/FlipCard/FlipCardBlock.tsx:98]: Check out if this crashes the frontend

File: services/course-material/src/components/ContentRenderer/moocfi/FlipCard/InnerCardBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+function isBlockImage(block: Block<unknown>): block is Block<FlipCardAttributes> {
+  return block.name === "core/image"
+}
+
+const InnerCardBlock: React.FC<React.PropsWithChildren<BlockRendererProps<FlipCardAttributes>>> = (
+  props,
+) => {
+  if (isBlockImage(props.data.innerBlocks[0]) && props.data.innerBlocks.length == 1) {
```
Comment (nygrenh) [inline services/course-material/src/components/ContentRenderer/moocfi/FlipCard/InnerCardBlock.tsx:28]: Add a comment for explaining the if

Comment (nygrenh) [reply services/course-material/src/components/ContentRenderer/moocfi/FlipCard/InnerCardBlock.tsx:28]: Does it crash if innerblocks is empty?

File: services/course-material/src/components/ContentRenderer/moocfi/FlipCard/FlipCardBlock.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    >
+      <div
+        className={css`
+          position: relative;
+          width: 100%;
+          height: 100%;
+          transition: transform 0.8s;
+          transform-style: preserve-3d;
+          ${frontSideUp ? "transform: rotateY(180deg);" : "transform: rotateY(0);"}
+          ${currentIsImage ? "" : "border: 3px solid #bfbec6"};
```
Comment (nygrenh) [inline services/course-material/src/components/ContentRenderer/moocfi/FlipCard/FlipCardBlock.tsx:70]:
```suggestion
          ${!currentIsImage && "border: 3px solid #bfbec6;"}
```

File: services/course-material/src/components/ContentRenderer/moocfi/FlipCard/FlipCardBlock.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+const FlipCardBlock: React.FC<React.PropsWithChildren<BlockRendererProps<FlipCardAttributes>>> = (
+  props,
+) => {
+  const { t } = useTranslation()
+  const frontCard = props.data.innerBlocks[0] as Block<FlipCardAttributes>
+  const backCard = props.data.innerBlocks[1] as Block<FlipCardAttributes>
+
+  let size = 0
```
Comment (nygrenh) [inline services/course-material/src/components/ContentRenderer/moocfi/FlipCard/FlipCardBlock.tsx:26]: create a function in this file sizeStringToSizePx

File: services/course-material/src/components/ContentRenderer/moocfi/FlipCard/FlipCardBlock.tsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+  return block.name === "core/image"
+}
+
+const FlipCardBlock: React.FC<React.PropsWithChildren<BlockRendererProps<FlipCardAttributes>>> = (
+  props,
+) => {
+  const { t } = useTranslation()
+  const frontCard = props.data.innerBlocks[0] as Block<FlipCardAttributes>
+  const backCard = props.data.innerBlocks[1] as Block<FlipCardAttributes>
+
```
Comment (nygrenh) [inline services/course-material/src/components/ContentRenderer/moocfi/FlipCard/FlipCardBlock.tsx:25]: Check if frontCard and backcard are defined and return an error banner if they are not.

File: services/cms/src/components/blocks/FlipCardSizeCustomizer.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
+/* eslint-disable i18next/no-literal-string */
```
Comment (nygrenh) [inline services/cms/src/components/blocks/FlipCardSizeCustomizer.tsx:1]: Don't need to ignore the whole file

**Review by nygrenh — CHANGES_REQUESTED** [review]
Lock the main blocks and add `templateLock={false}` to the inner inner blocks.

**Review by nygrenh — APPROVED** [review]

**Pull request merged**